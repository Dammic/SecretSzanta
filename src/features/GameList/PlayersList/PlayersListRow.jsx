import React from 'react'
import PropTypes from 'prop-types'

const PlayersListRow = ({ playerName, avatarNumber }) => {

    const avatarPicture = require(`../../../static/Avatar${avatarNumber}.png`)
    return (
        <div>
            <img className="avatar" src={avatarPicture} alt="Player avatar" />
            <span></span>
            <span>{playerName}</span>
        </div>
    )
}

PlayersListRow.propTypes = {
    playerName: PropTypes.string.isRequired,
    avatarNumber: PropTypes.number.isRequired,
}

export default PlayersListRow
