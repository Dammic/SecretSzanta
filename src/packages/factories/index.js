export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
