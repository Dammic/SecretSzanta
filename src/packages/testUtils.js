import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
