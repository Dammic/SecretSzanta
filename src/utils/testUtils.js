import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
