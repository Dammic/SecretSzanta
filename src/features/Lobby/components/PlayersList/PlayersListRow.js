import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { getAvatar } from '../../../../utils/avatarsHelper'

import styles from './PlayersList.module.css'
import commonStyles from '../../../Shared/CommonStyles/commonStyles.module.css'

export const PlayersListRow = ({ playerName, avatarNumber, currentRoom }) => {
    return (
        <div className={classNames(styles.playerRow, { [styles.busy]: !!currentRoom })}>
            <img
                className={styles.avatar}
                src={getAvatar(`liberal-${avatarNumber}`)}
                alt="Player avatar"
            />
            <span className={classNames(commonStyles.ellipsis, styles.playerName)}>{playerName}</span>
            {currentRoom && (
                <Fragment>
                    <span className={classNames(commonStyles.ellipsis, styles.roomName)}>
                        <span>
                            <span>room:{' '}</span>
                            <b className={styles.roomNameText}>{currentRoom}</b>
                        </span>
                    </span>
                    <span className={classNames(styles.roomName, styles.mobile)}>Busy</span>
                </Fragment>
            )}
        </div>
    )
}

PlayersListRow.propTypes = {
    playerName: PropTypes.string.isRequired,
    avatarNumber: PropTypes.number.isRequired,
    currentRoom: PropTypes.string,
}
