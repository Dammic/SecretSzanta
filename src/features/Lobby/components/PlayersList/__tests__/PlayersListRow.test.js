import React from 'react'
import { shallow } from 'enzyme'
import { expectShallowMatchingSnapshot } from 'packages/testUtils'
import { PlayersListRow } from '../PlayersListRow'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        playerName: 'mockPlayerName',
        avatarNumber: 1,
        currentRoom: 'mockCurrentRoom',
        ...propsOverrides,
    }
    const component = renderMethod(<PlayersListRow {...props} />)
    return { props, component }
}

describe('<PlayersListRow />', () => {
    it('matches snapshot - renders user with current room', () => {
        const { props } = setupProps()
        expectShallowMatchingSnapshot(<PlayersListRow {...props} />)
    })

    it('matches snapshot - renders user without current room', () => {
        const { props } = setupProps({ currentRoom: null })
        expectShallowMatchingSnapshot(<PlayersListRow {...props} />)
    })
})
