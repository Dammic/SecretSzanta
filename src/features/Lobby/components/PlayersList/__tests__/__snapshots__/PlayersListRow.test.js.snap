// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayersListRow /> matches snapshot - renders user with current room 1`] = `
<div
  className="playerRow busy"
>
  <img
    alt="Player avatar"
    className="avatar"
    src="test-file-stub"
  />
  <span
    className="ellipsis playerName"
  >
    mockPlayerName
  </span>
  <React.Fragment>
    <span
      className="ellipsis roomName"
    >
      <span>
        <span>
          room:
           
        </span>
        <b
          className="roomNameText"
        >
          mockCurrentRoom
        </b>
      </span>
    </span>
    <span
      className="roomName mobile"
    >
      Busy
    </span>
  </React.Fragment>
</div>
`;

exports[`<PlayersListRow /> matches snapshot - renders user without current room 1`] = `
<div
  className="playerRow"
>
  <img
    alt="Player avatar"
    className="avatar"
    src="test-file-stub"
  />
  <span
    className="ellipsis playerName"
  >
    mockPlayerName
  </span>
</div>
`;
